src/java.base/share/classes/sun/util/locale/LocaleMatcher.java

```diff
@@ -124,8 +124,16 @@
         for (LanguageRange lr : nonZeroRanges) {
             String range = lr.getRange();
             if (range.equals("*")) {
-                tags = removeTagsMatchingBasicZeroRange(zeroRanges, tags);
-                return new ArrayList<String>(tags);
+                for (String tag : tags) {
+                    String lowerCaseTag = tag.toLowerCase(Locale.ROOT);
+
+                    if (!caseInsensitiveMatch(list, lowerCaseTag)
+                            && !shouldIgnoreFilterBasicMatch(zeroRanges, lowerCaseTag)) {
```

Comment:
Is there any reason to flip the evaluation order of the `if` statement?

Reply:
Two reasons:
  - It's consistent with the `else`  branch in the `for` loop of `LocaleMatcher.{filterBasic,filterExtended}` (main)
  - I thought that `caseInsensitiveMatch` being first made it clear the need for `lowerCaseTag` but now noticed that the comment was meant for `{shouldIgnoreFilterBasicMatch,shouldIgnoreFilterExtendedMatch}` instead